### PR TITLE
Refactor using SOLID design

### PR DESCRIPTION
**Pull request by MrBrN197**
## Changes
- Refactor `App` to its own file.
- Move input handling functionality to `main.rb`
- Create `BookStorage` class to store and handle book functionality
- Create `PersonStorage` class to store and handle people functionality

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    puts
+    print 'Date: '
+    date = gets.chomp
+
+    rental = Rental.new(date, @person_storage.get_person_at_index(person_i), @book_storage.get_book_at_index(book_i))
+    @rentals.push(rental)
+
+    puts 'Rental created successfullyy'
+  end
```
Comment (mikethreels) [inline app.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed to the console information is gathered and the gathered information is processed, this is not according to the single responsibility rule, try to split this in smaller easier to manage methods. for reference have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

File: book_storage.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books.push(book)
+
+    puts 'Book created successfully'
+  end
```
Comment (mikethreels) [inline book_storage.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed to the console information is gathered and the gathered information is processed, this is not according to the single responsibility rule, try to split this in smaller easier to manage methods. for reference have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

File: person_storage.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+
+    student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
+    @people.push(student)
+
+    puts 'Person created successfully'
+  end
```
Comment (mikethreels) [inline person_storage.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed to the console information is gathered and the gathered information is processed, this is not according to the single responsibility rule, try to split this in smaller easier to manage methods. for reference have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

File: person_storage.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age: age, name: name, specialization: specialization)
+    @people.push(teacher)
+
+    puts 'Person created successfully'
+  end
```
Comment (mikethreels) [inline person_storage.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed to the console information is gathered and the gathered information is processed, this is not according to the single responsibility rule, try to split this in smaller easier to manage methods. for reference have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Orora, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._
- [x] in your readme file under the title Getting Started you mention the following command `cd git clone https://github.com/wuyepabdul/school-library` this doesn't work, it should be without `cd` in the beginning.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person_storage.rb (hunk 4 above)
Comment (MrBrN197) [reply person_storage.rb]: Thank, you I've addressed this

File: person_storage.rb (hunk 3 above)
Comment (MrBrN197) [reply person_storage.rb]: Thank, you I've addressed this

File: book_storage.rb (hunk 2 above)
Comment (MrBrN197) [reply book_storage.rb]: Thank, you I've addressed this

File: app.rb (hunk 1 above)
Comment (MrBrN197) [reply app.rb]: Thank, you I've addressed this

File: book_storage.rb (hunk 2 above)
Comment (MrBrN197) [reply book_storage.rb]: Thank You, I've addressed this issue.

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**